lib/active_force/sobject.rb

```diff
@@ -34,8 +34,24 @@ class << self
       def_delegators :query, :not, :or, :where, :first, :last, :all, :find, :find!, :find_by, :find_by!, :sum, :count, :includes, :limit, :order, :select, :none
       def_delegators :mapping, :table, :table_name, :custom_table?, :mappings
 
+      def update(id, attributes)
+        prepare_for_update(id, attributes).update
+      end
+
+      def update!(id, attributes)
+        prepare_for_update(id, attributes).update!
+      end
+
       private
 
+      def prepare_for_update(id, attributes)
+        new(attributes.merge(id: id)).tap do |obj|
+          attributes.each do |name, value|
+            obj.public_send("#{name}_will_change!") if value.nil?
```

Comment:
I think this could be generalized to:

```ruby
obj.public_send("#{name}_will_change!") unless object.public_send("#{name}_changed?")
```

But this is more confusing and I think we should avoid those `changed?` calls if it's not really needed.